spring-integration-core/src/main/java/org/springframework/integration/handler/advice/RateLimiterRequestHandlerAdvice.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.handler.advice;
+
+import java.time.Duration;
+
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessagingException;
+import org.springframework.util.Assert;
+
+import io.github.resilience4j.ratelimiter.RateLimiter;
+import io.github.resilience4j.ratelimiter.RateLimiterConfig;
+import io.github.resilience4j.ratelimiter.RequestNotPermitted;
+import io.vavr.CheckedFunction0;
+import io.vavr.control.Try;
+
+/**
+ * An {@link AbstractRequestHandlerAdvice} extension for a rate limiting to service method calls.
+ * The implementation is based on the
+ * <a href="https://github.com/resilience4j/resilience4j#ratelimiter">Resilience4j</a>.
+ *
+ * @author Artem Bilan
+ *
+ * @since 5.2
+ */
+public class RateLimiterRequestHandlerAdvice extends AbstractRequestHandlerAdvice {
+
+	public static final String DEFAULT_NAME = "RateLimiterRequestHandlerAdvice";
+
+	private final RateLimiter rateLimiter;
+
+	private ApplicationEventPublisher applicationEventPublisher;
```

Comment:
Unused field.

Reply:
Yeah... I was thought to propagate `RateLimiter` event to Spring events, but then I realized that their API allows only to have one event consumer, so it is going to be some kind of abuse to mutate externally provided `RateLimiter`.
Om the other hand their even doesn't bring too much info to be propagated with our own events.
On the other hand we are good to live without them with just call we have here.
Plus that's not too hard to propagate those events by yourself configuring `RateLimiter` externally as a bean.